### DISCUSSION TITLE
Idea, spam account block list

### DISCUSSION THREAD

**Discussion by ageekhere** [ThioJoe/YT-Spammer-Purge#38] [Ideas]
I wonder if a community run spammer account list could be created. People could submit known bots to the list and have their messages auto deleted.

**Comment by MDP43140** [comment]
@ageekhere yeah that's good idea

**Comment by avighnac** [comment]
That is a good idea but new spam accounts will keep being created.

**Comment by joaogelado** [comment]
> That is a good idea but new spam accounts will keep being created. 

But, before they figure out, it would be very effective

**Comment by avighnac** [comment]
True. This should be done.

**Comment by ageekhere** [comment]
Maybe the script could even have an option to ban the bots that are in the list as well as deleting all their comments.

**Comment by deidyomega** [comment]
If there is interest and @ThioJoe is cool with it, I could add that functionality

**Comment by TechStudent10** [comment]
Yeah, like having a website to submit it to. Then, use an API to request for that list. I could actually implement this. Of course, I would need to look through ThioJoe's code and understand it more to implement this feature, as well as create an entire website with an API to submit it to. Yeah, definitely a weekend project.

**Comment by ThioJoe** [comment]
The issue is it might massively slow down the program the more that get added. Python is not exactly known for being fast. I guess I could experiment with having it check huge numbers of channel IDs.

**Reply by ageekhere** [reply]
Maybe using multiple threads could help.

**Reply by TechStudent10** [reply]
@ageekhere @ThioJoe Yeah, just check for more spam channel IDs in a separate thread while the main program does stuff.

**Reply by ryanolee** [reply]
Python sets have a constant time lookup complexity. Loading all the ID's into one then using an in operator should result in no noticeable slowdown only higher memory usage depending on the size of the filter https://wiki.python.org/moin/TimeComplexity 🤔

**Reply by deidyomega** [reply]
I dont think python speed would really be at play here, you could have a set with 50k id's within it, and you wouldn't even notice the difference.  Using the exact same program structure you have, but instead of checking against one id, use the `in` statement.

**Comment by ageekhere** [comment]
I wonder if the script could somehow submit spam accounts to a community ran block list (which the script could update from). That way the block list could auto update itself from user submissions.
Some logic could be added to verify if a submitted account is not a false positive.  The script could check for well know spam accounts and delete all messages from them while also reporting them until they are removed.

Maybe even a telly could be made to see how many were shutdown over time, (e,g 243,000 spam accounts have been shutdown).
